addons/website_profile/controllers/main.py

```diff
@@ -91,14 +92,28 @@ def get_user_profile_avatar(self, user_id, field='avatar_256', width=0, height=0
             field_name=field, width=int(width), height=int(height), crop=crop
         ).get_response()
 
+    def _prepare_url_from(self):
+        url_from = request.httprequest.headers.get('Referer')
+        no_url = {'url_from_label': None, 'url_from': None}
+        if url_from and ((url_from_parsed := urlsplit(url_from)).netloc == urlsplit(request.httprequest.url).netloc):
+            path = url_from_parsed.path
+            return next(({'url_from_label': label, 'url_from': url_from}
+                         for prefix, label in (('forum', _('Forum')), ('slides', _('All Courses')))
+                         if path == f'/{prefix}' or path.startswith(f'/{prefix}/')),
+                        no_url)
```

Comment:
Maybe some linting could help reading the generator
```suggestion
            return next(
                (
                    {'url_from_label': label, 'url_from': url_from}
                    for prefix, label in (('forum', _('Forum')), ('slides', _('All Courses')))
                    if path == f'/{prefix}' or path.startswith(f'/{prefix}/')
                ), no_url)
```